rust/src/lib.rs

```diff
@@ -522,6 +522,7 @@ pub mod ffi {
             checksum_from: &str,
             checksum_to: &str,
         ) -> Result<*mut GVariant>;
+        fn generate_object_path(base: &str, segments: &[&str]) -> Result<String>;
```

Comment:
This possibly better belongs to `daemon.rs`, as it only used by the daemon logic.